src/propertySelector/propertySelector.js

```diff
@@ -0,0 +1,150 @@
+import component from 'ui/component';
+import propertySelectorTpl from 'ui/propertySelector/tpl/property-selector';
+import propertyDescriptionTpl from 'ui/propertySelector/tpl/property-description';
+import checkBoxTpl from 'ui/dialog/tpl/checkbox';
+import buttonFactory from 'ui/button';
+import 'ui/propertySelector/css/propertySelector.css';
+import $ from 'jquery';
+
+export default function propertySelectorFactory(config) {
+    //element references
+    let $container;
+    let $buttonsContainer;
+    let $propertyListContaner;
+    let $searchInput;
+    let availableProperties = [];
+    let selectedProperties;
+    let search = '';
+
+    const parentGap = 20;
+
+    /**
+     * Lookup for characters in text to highlight
+     * @param {String} text - text to lookup
+     * @param {String} highlight - character(s) to be highlighted
+     * @param {regExp|String} match - match to be applied in the text
+     * @returns {String} - highlighted text
+     */
+    function highlightCharacter(text, highlight, match) {
+        return text.replace(match, `<b>${highlight}</b>`);
+    }
+
+    /**
+     * Creates property description list element
+     * @param {*} property
+     * @returns JQuery element containing property description
+     */
+    function createPropertyOption(property) {
+        const descriptionData = Object.assign({}, property);
+        if (search !== '') {
+            descriptionData.label = highlightCharacter(descriptionData.label, search, search);
+        }
+        const $propertyDescription = $(propertyDescriptionTpl({ property: descriptionData }));
+        const $checkboxContainer = $('.checkbox-container', $propertyDescription);
+        $checkboxContainer.append(checkBoxTpl({ id: descriptionData.id, checked: descriptionData.selected }));
+        const $checkbox = $('input', $checkboxContainer);
+        $checkbox.on('change', function () {
+            if (this.checked) {
+                selectedProperties.add(property.id);
+            } else {
+                selectedProperties.delete(property.id);
+            }
+        });
+        return $propertyDescription;
+    }
+
+    const instance = component({
+        positionContainer: function positionContainer() {
+            let { top, left, right, bottom } = this.config.data.position;
+            let maxHeight;
+            if (typeof bottom === 'undefined') {
+                maxHeight = $container.parent().height() - top - parentGap;
+            }
+            if (typeof top === 'undefined') {
+                maxHeight = $container.parent().height() - bottom - parentGap;
+            }
+            $container.css({ top, left, right, bottom, maxHeight });
```

Comment:
**question:** is it needed? Could not this be managed directly via CSS?

Reply:
I did not find a way to hanle it through css

Reply:
Well, considering the current use case, I think it is a bit over-engineered as the component will always be placed on the top. Anyway, not a big issue in the end.